packages/theme/pages/Product.vue

```diff
@@ -470,61 +462,76 @@ export default defineComponent({
       });
     };
 
-    const updateProductConfiguration = async (label, value) => {
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      productConfiguration.value.push([label, value]);
+    let firstFetch = true;
+    const { fetch } = useFetch(async () => {
+      const { params: { id } } = route.value;
 
-      await router.push({
-        path: `${app.localePath(route.value.fullPath)}`,
-        query: {
-          ...Object.fromEntries(productConfiguration.value),
-        },
-      });
-    };
-
-    useAsync(async () => {
       const baseSearchQuery = {
         filter: {
           sku: {
             eq: id,
           },
         },
-        ...(productConfiguration.value.length > 0
-          ? {
-            configurations: productConfiguration.value.map(
-              (config) => config[1],
-            ),
-          }
-          : {}),
+        configurations: productConfiguration.value.map(
+          (config) => config[1],
+        ) as string[],
       };
 
-      products.value = await getProductDetails({
+      const result = await getProductDetails({
         ...baseSearchQuery,
       });
 
-      if (product?.value?.length === 0) nuxtError({ statusCode: 404 });
+      product.value = result.items[0] as Product ?? null;
+      if (Boolean(product?.value?.sku) === false) nuxtError({ statusCode: 404 });
 
-      productReviews.value = await searchReviews(baseSearchQuery);
+      if (firstFetch) {
+        firstFetch = false;
+        const productReviews = await searchReviews(baseSearchQuery);
 
-      const tags = [
-        {
-          prefix: CacheTagPrefix.View,
-          value: `product-${route.value.params.id}`,
-        },
-      ];
-      const productTags = [{
-        prefix: CacheTagPrefix.Product,
-        value: product.value.uid,
-      }];
-
-      const categoriesTags = categories.value.map((catId) => ({
-        prefix: CacheTagPrefix.Category,
-        value: catId,
-      }));
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      addTags([...tags, ...productTags, ...categoriesTags]);
+        const baseReviews = Array.isArray(productReviews)
+          ? [...productReviews].shift()
```

Comment:
```suggestion
          ? productReviews[0]
```

I know this is just a diff because you moved the code from above but this is so weird that I felt the need to leave a comment.

shift() removes the first element from the array (mutates the array) and returns the first element. So here a new array is created just for the sake of getting the first element from it. This doesn't even create any new references or anything

Reply:
ok